cpp/src/arrow/engine/substrait/relation_internal.cc

```diff
@@ -49,6 +50,59 @@ Status CheckRelCommon(const RelMessage& rel) {
   return Status::OK();
 }
 
+Result<fs::FileInfoVector> GetGlobFiles(const std::shared_ptr<fs::FileSystem>& filesystem,
+                                        std::string& path) {
+  fs::FileInfoVector results, temp;
+  fs::FileSelector selector;
+  std::string cur;
+  size_t i = 0;
+
+#if _WIN32
+  auto split_path = fs::internal::SplitAbstractPath(path, '\\');
+  ARROW_ASSIGN_OR_RAISE(auto file, filesystem->GetFileInfo(split_path[i++] + "\\"));
+#else
+  auto split_path = fs::internal::SplitAbstractPath(path, '/');
+  ARROW_ASSIGN_OR_RAISE(auto file, filesystem->GetFileInfo("/"));
```

Comment:
Should this be `split_path[i++] + "/"`?

Reply:
I was thinking in this glob "/*/A/a.txt" in linux local filesystem, so the first element should be "/"